app/views/candidate_interface/application_form/complete.html.erb

```diff
@@ -1,8 +1,11 @@
 <% content_for :title, t('page_titles.application_dashboard') %>
 
 <% new_references_needed_component = CandidateInterface::NewReferencesNeededComponent.new(application_form: @application_form) %>
+<% new_application_choice_needed_component = CandidateInterface::NewCourseChoiceNeededBannerComponent.new(application_form: @application_form) %>
 <% if new_references_needed_component.render? %>
   <%= render new_references_needed_component %>
+<% elsif FeatureFlag.active?('replace_full_or_withdrawn_application_choices') && new_application_choice_needed_component.render? %>
```

Comment:
Instead of calling `.render?` on the instantiated component, you could use it inside the template and render nothing if it's false. I think that's better for encapsulation.

Reply:
Yea good shout 👌